internal/venice-common/src/main/java/com/linkedin/venice/controller/requests/ControllerHttpRequest.java

```diff
@@ -0,0 +1,80 @@
+package com.linkedin.venice.controller.requests;
+
+import com.linkedin.venice.controllerapi.QueryParams;
+import com.linkedin.venice.utils.Time;
+
+
+/**
+ * A general class to represent Http specific transport request that embodies all the Http requests
+ * send to the controller as part of controller APIs. This is a container class defined to help with the
+ * refactoring of ControllerClient to become transport protocol agnostic.
+ */
+public class ControllerHttpRequest {
```

Comment:
What is the relationship between this class and `ControllerRequest`? Should this one extend the other?

Reply:
`ControllerRequest` abstraction is meant to decouple the actual transport related information. `ControllerHttpRequest` however, describes how `ControllerRequest` will look like if http was chosen as the medium of transport.

Originally, I thought about having some coupling between `ControllerRequest`, `ControllerHttpRequest` and `GeneratedV3..` (grpc general request type)in the form of having `ControllerRequest` host few abstract methods like 

```
  public abstract ControllerRequest {
      ...
      abstract ControllerHttpRequest convertToHttpRequest();

      abstract <T extends GeneratedV3..> T convertToGrpcRequest();
 }
```

However, the above spills some of the transport details into the model and hence chose against it;

Reply:
Hmm... but isn't it Protobuf's design goal to provide a one stop shop for models and their transport? And isn't that also what we do with Jackson (i.e. the model classes have transport-related annotations sprinkled over them). What is the interest in decoupling model from transport? (I understand the purpose of decoupling in the abstract, but I am asking specifically in this context, what does it buy us?)